File: packages/api-client/src/doubts.ts
import { apiClient } from './client';
import type { Doubt } from '@ai-tutor/types';

export const doubtsApi = {
  async create(
    lesson_id: string,
    question: string,
    video_timestamp?: number,
    context?: string
  ): Promise<Doubt> {
    const response = await apiClient.post<Doubt>('/api/doubt', {
      lesson_id,
      question,
      video_timestamp,
      context,
    });
    return response.data;
  },
};

File: apps/api/routers/doubts.py
from fastapi import APIRouter
from pydantic import BaseModel
from datetime import datetime

router = APIRouter()


class DoubtRequest(BaseModel):
    lesson_id: str
    question: str
    video_timestamp: float = None
    context: str = None


@router.post("/")
async def create_doubt(request: DoubtRequest):
    """Create a doubt - placeholder"""
    return {
        "id": "doubt-123",
        "lesson_id": request.lesson_id,
        "question": request.question,
        "answer": f"Great question about '{request.question}'. This is a placeholder response.",
        "created_at": datetime.now().isoformat()
    }
